rosette/src/BigBang.cc

```diff
@@ -380,72 +380,58 @@ void InitBuiltinObs() {
 }
 
 
-static void get_path_prefix(char* path, char* dir) {
-    char* p = strrchr(path, '/');
-    if (p) {
-        int n = p - path;
-        strncpy(dir, path, n);
-        dir[n] = 0;
-    } else {
-        strcpy(dir, ".");
-    }
-}
-
-
 const char* StandardExtensions[] = {".rbl", 0};
 
 
 static FILE* FindBootFile() {
     char path[MAXPATHLEN];
-    char* RosetteLib = getenv("ROSETTE_LIB");
 
-    if (strcmp(BootFile, "") == 0) {
-        if (RosetteLib) {
-            strcpy(BootDirectory, RosetteLib);
-        }
+    strcpy(path, BootDirectory);
+    strcat(path, "/");
```

Comment:
Can't be sure we're not overrunning buffer here. Suggest something like
```C++
    auto bootfile = BootFile;
    if (*bootfile == '\0') {
        bootfile = "boot.rbl";
    }

    snprintf(path, MAXPATHLEN, "%s/%s", BootDirectory, bootfile);

    // ...
```

Reply:
Yes, that's better. Done.